### PR TITLE
Remove user-management error cases from scenario-service proto.

### PR DESCRIPTION
**Pull request by nickchapman-da**
Since the expected user-management errors are now reified within `ScriptLedgerClient`, remaining errors are unexpected.
#11997

### CODE REVIEW

File: compiler/scenario-service/server/src/main/scala/com/digitalasset/daml/lf/scenario/Conversions.scala [hunk 1]
```diff
@@ -251,12 +251,7 @@ final class Conversions(
         builder.setScenarioPartyAlreadyExists(party)
 
       case Error.UserManagement(err) =>
```
Comment (cocreature) [inline compiler/scenario-service/server/src/main/scala/com/digitalasset/daml/lf/scenario/Conversions.scala:253]: Do we need this error case in the Scala AST? It seems like we should never hit this now?

**Review by cocreature — APPROVED** [review]
thanks!

File: compiler/scenario-service/server/src/main/scala/com/digitalasset/daml/lf/scenario/Conversions.scala (hunk 1 above)
Comment (nickchapman-da) [reply compiler/scenario-service/server/src/main/scala/com/digitalasset/daml/lf/scenario/Conversions.scala]: I removed it.

Comment (nickchapman-da) [reply compiler/scenario-service/server/src/main/scala/com/digitalasset/daml/lf/scenario/Conversions.scala]: In fact we now don't need a type for `UserManagementError` in daml-script at all... It remained used only to communicate from `UserManagementStore` to `IdeLedgerClient`, where it was reduced anyway to just `Option`.

I changed to have the same `Option` based interface also in `UserManagementStore`.

And code just gets removed!

**Review by cocreature — COMMENTED** [review]
looks like some of the tests need fixing

**Pull request merged**